test/PHPMailer/PHPMailerTest.php

```diff
@@ -1191,6 +1191,46 @@ public function testDuplicateIDNRemoved()
         );
     }
 
+    /**
+     * Test SMTPUTF8 usage, including when it is not to be used.
+     */
+    public function testUnsuppoortedSmtpUTF8()
+    {
+        $this->Mail->CharSet = PHPMailer::CHARSET_ISO88591;
+	self::assertFalse($this->Mail->addAddress('motörhead@example.com', ''));
+    }
+
+    /**
+     * Test SMTPUTF8 usage, including when it is not to be used.
+     */
+    public function testSmtpUTF8()
+    {
+        //No reason to use SMTPUTF8
+        $this->Mail->isSMTP();
+        $this->Mail->addAddress('foo@example.com', '');
+        $this->Mail->preSend();
+
+        //Using a punycoded domain is enough
+        self::assertFalse($this->Mail->UseSMTPUTF8);
+        $this->Mail->addAddress('foo@motörhead.example', '');
```

Comment:
I like to avoid characters like ö for testing UTF-8 support because they are also valid in ISO8859 charsets. The classic example is `Iñtërnâtiônàlizætiøn`, which is quite useless as all of its chars are also in 8859-1! Maybe something radically non-ISO, like 😱?

Reply:
Let me know if you need some more test data (or have a look here for some samples you can build on: https://github.com/PHPCSStandards/PHPCSUtils/blob/develop/Tests/Utils/Orthography/FirstCharTest.php)

Talking about the tests - those could maybe do with some expanding / these methods getting their own test file ? With a `@covers` tag too ?

Reply:
What, you're not a Motörhead fan? I'll change to Spın̈al Tap! The n̈ is in unicode, but not in any 8859 variant.

I'd rather not use emoji or anything else that's blocked in public TLDs, bad examples leads to FAQs for me at work. (Emoji are banned in general, because second-level domains should be distinguishable and many emoji pairs look confusingly similar.)